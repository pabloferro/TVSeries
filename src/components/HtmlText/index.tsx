import React from 'react';
import {useWindowDimensions} from 'react-native';
import RenderHtml from 'react-native-render-html';
import {white} from '../../constants/colors';

export default function HtmlText({children}: {children: string}) {
  const {width} = useWindowDimensions();
  return (
    <RenderHtml
      contentWidth={width}
      source={{html: children}}
      // eslint-disable-next-line react-native/no-inline-styles
      baseStyle={{color: white, fontSize: 16, fontWeight: '400'}}
    />
  );
}
