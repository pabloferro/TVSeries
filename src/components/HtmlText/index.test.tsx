import React from 'react';
import {render} from '@testing-library/react-native';

import HtmlText from '.';

test('renders correctly with children', () => {
  const {toJSON} = render(<HtmlText>{'<p>some text</p>'}</HtmlText>);
  expect(toJSON()).toMatchSnapshot();
});

test('renders nothing when children is undefined', () => {
  const {toJSON} = render(<HtmlText>{undefined}</HtmlText>);
  expect(toJSON()).toMatchSnapshot();
});
