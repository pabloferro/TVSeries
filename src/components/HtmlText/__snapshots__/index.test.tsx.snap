// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with children 1`] = `
<View
  style={
    Array [
      Object {},
      undefined,
      undefined,
    ]
  }
  testID="html"
>
  <View
    style={
      Array [
        Object {},
        undefined,
        undefined,
      ]
    }
    testID="body"
  >
    <View
      style={
        Array [
          Object {
            "marginBottom": 14,
            "marginTop": 14,
          },
          undefined,
          undefined,
        ]
      }
      testID="p"
    >
      <Text
        allowFontScaling={true}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#FFF",
              "fontSize": 16,
              "fontWeight": "400",
            },
            undefined,
            undefined,
          ]
        }
      >
        some text
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders nothing when children is undefined 1`] = `null`;
